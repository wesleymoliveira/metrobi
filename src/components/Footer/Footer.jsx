import React from 'react'
import * as S from './Footer.styled'
import { Typography } from '../Typography/Typography';

export const Footer = () => (
  <S.Wrapper data-testid='footer'>
    <Typography size='xxlarge'>
      Footer
    </Typography>
  </S.Wrapper>
);
