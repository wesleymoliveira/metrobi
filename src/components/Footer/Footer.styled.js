import styled, { css } from 'styled-components'

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.footerBackground};
    height: 8rem;
    display: flex;
    justify-content: center;
    align-items: center;
  `}
`;
