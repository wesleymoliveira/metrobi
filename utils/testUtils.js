import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
