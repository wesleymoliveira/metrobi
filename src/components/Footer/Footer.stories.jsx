import { Footer } from "./Footer"


export default {
  title: 'components/Footer',
  component: Footer,
}

export const Default= () => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <Footer />
  </div>
)
