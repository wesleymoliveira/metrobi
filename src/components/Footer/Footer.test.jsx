import { screen } from '@testing-library/react';
import { render, snapshotRenderer } from '../../../utils/testUtils';

import { Footer } from './Footer';

describe('<Footer />', () => {
		it('should render the component', () => {
		render(<Footer />);
		
		expect(screen.getByTestId(/footer/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Footer />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<Footer />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#FFA500')
	});

});