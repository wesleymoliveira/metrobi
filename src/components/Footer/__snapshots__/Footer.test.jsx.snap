// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
.c0 {
  background-color: #FFA500;
  height: 8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

<footer
  className="c0"
  data-testid="footer"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Footer
  </h1>
</footer>
`;
